lua/spectre_common.lua

```diff
@@ -105,25 +127,39 @@ local function determine_if_cacheable(url, namespace, request_headers)
                 refresh_cache = false,
             }
 
-            -- Only cache GET and HEAD requests
-            local http_method = ngx.req.get_method()
-            if http_method ~= 'GET' and http_method ~= 'HEAD' then
-                cacheability_info.is_cacheable = false
-                cacheability_info.reason = 'non-cacheable-method'
-            end
-
             -- Check if client sent no-cache header
             if has_marker_headers(request_headers, PLEASE_DO_NOT_CACHE_HEADERS) then
                 cacheability_info.is_cacheable = false
                 cacheability_info.reason = 'no-cache-header'
                 cacheability_info.refresh_cache = true
             end
-            break
+
+            if http_method == 'POST' then
+                if not has_marker_headers(request_headers, POST_CACHEABLE_HEADERS) then
+                    -- For Post requests check the content type is application/json
+                    cacheability_info.is_cacheable = false
+                    cacheability_info.reason = 'non-cacheable-content-type'
+                    cacheability_info.refresh_cache = false
```

Comment:
isn't this the default?

Reply:
no. if the pattern and request method matches, we mark it as cacheable, and check different conditions, to set is_cacheable to False.